testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -93,3 +94,64 @@ def test_incomplete_defined_analysis(self):
 
     def test_old_api(self):
         OldAPIAnalysis(self.u.trajectory).run()
+
+
+def test_filter_baseanalysis_kwargs():
+    def bad_f(mobile, step=2):
+        pass
+
+    def good_f(mobile, ref):
+        pass
+
+    kwargs = {'step': 3, 'foo': None}
+
+    assert_raises(ValueError, base._filter_baseanalysis_kwargs, bad_f, kwargs)
+
+    base_kwargs, kwargs = base._filter_baseanalysis_kwargs(good_f, kwargs)
+
+    assert_equal(1, len(kwargs))
+    assert_equal(kwargs['foo'], None)
+
+    assert_equal(4, len(base_kwargs))
+    assert_equal(base_kwargs['start'], None)
+    assert_equal(base_kwargs['step'], 3)
+    assert_equal(base_kwargs['stop'], None)
+    assert_equal(base_kwargs['quiet'], True)
+
+
+def simple_function(mobile):
+    return mobile.center_of_geometry()
+
+
+def test_AnalysisFromFunction():
+    u = mda.Universe(PSF, DCD)
+    step = 2
+    ana = base.AnalysisFromFunction(simple_function, mobile=u.atoms,
+                                    step=step).run()
+
+    ana = base.AnalysisFromFunction(simple_function, u.atoms,
```

Comment:
Then, maybe should we make sure that they give the same result. Maybe they could be saved in `ana1` and `ana2`, which could both be checked against the reference.